corehq/ex-submodules/pillowtop/pillow/interface.py

```diff
@@ -100,29 +101,101 @@ def run(self):
         self.process_changes(since=self.get_last_checkpoint_sequence(), forever=True)
 
     def _update_checkpoint(self, change, context):
-        if change:
+        if change and context:
             updated = self.fire_change_processed_event(change, context)
         else:
             updated = self.checkpoint.touch(min_interval=CHECKPOINT_MIN_WAIT)
         if updated:
             self._record_checkpoint_in_datadog()
 
+    @property
+    def _should_process_in_chunks(self):
+        # chunked processing is only supported for pillows with single processor
```

Comment:
Why is this necessary?

Reply:
It's not a hard requirement. I kept this constraint to keep the `process_changes` logic (pertaining to checkpoint updates and multiple processors having different `processor_chunk_size`) simple and also considering that there is only one pillow that has multiple processors right now (and none of them a bulk processor). It's not difficult to get rid of this constraint, but it wasn't necessary when I started the PR